extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -727,8 +740,8 @@ private OutputChannelFactory makeSuperSorterIntermediateOutputChannelFactory(
     final FileOutputChannelFactory fileOutputChannelFactory =
         new FileOutputChannelFactory(fileChannelDirectory, frameSize, intermediateSuperSorterLocalStorageTracker);
 
-    if (MultiStageQueryContext.isComposedIntermediateSuperSorterStorageEnabled(QueryContext.of(task.getContext()))
```

Comment:
We should only use the composing output channel if the `tmpStorageBytesPerTask` is < Long.MAX value.
That way people who are using durable storage but have not configured `druid.indexer.task.tmpStorageBytesPerTask` go through the same code path.

Reply:
Changed to only trigger when the value is explicitly set in the run time parameters